tests/e2e/deployment-test.sh

```diff
@@ -0,0 +1,145 @@
+  if [[ "${CONTAINER_STATE}" != *"running"* ]]; then
+    return 1
+  fi
+  return 0
+}
+
+while (( "$#" )); do
+    case "$1" in
+            "--logs")
+                DEPLOYMENT_TYPE="logs"
+                shift
+            ;;
+
+            "--metrics")
+                DEPLOYMENT_TYPE="metrics"
+                shift
+            ;;
+
+            "--all")
+                DEPLOYMENT_TYPE="all"
+                shift
+            ;;
+
+            *)
+            echo "Unknown param $1"
+            print_help
+            exit 1
+    esac
+done
+
+
+
+# Install kubectl.
+curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl" && chmod +x kubectl && sudo mv kubectl /usr/local/bin/kubectl
+
+# Install helm.
+wget https://get.helm.sh/helm-v3.5.3-linux-amd64.tar.gz
+FILE=./helm-v3.5.3-linux-amd64.tar.gz
+if [ ! -f "$FILE" ]; then
+    echo "$FILE does not exist. Can't create helm chart."
+    exit 1
+fi
+tar -zxvf helm-v3.5.3-linux-amd64.tar.gz
+FILE=./linux-amd64/helm
+if [ ! -f "$FILE" ]; then
+    echo "$FILE does not exist. Can't create helm chart."
+    exit 1
+fi
+
+sudo mv linux-amd64/helm /usr/local/bin/helm
+
+
+# Create Pub/Sub topic and subscription.
+gcloud config set project dynatrace-gcp-extension
+
+if [[ $(gcloud pubsub topics list --filter=name:"${PUBSUB_TOPIC}" --format="value(name)") ]]; then
+    echo "Topic [${PUBSUB_TOPIC}] already exists, skipping"
+else
+    gcloud pubsub topics create "${PUBSUB_TOPIC}"
+fi
+
+if [[ $(gcloud pubsub subscriptions list --filter=name:"${PUBSUB_SUBSCRIPTION}" --format="value(name)") ]]; then
+    echo "Subscription [${PUBSUB_SUBSCRIPTION}] already exists, skipping"
+else
+    gcloud pubsub subscriptions create "${PUBSUB_SUBSCRIPTION}" --topic="${PUBSUB_TOPIC}" --ack-deadline=120
+fi
+
+
+# Run helm deployment.
+rm -rf ./e2e_test
+mkdir ./e2e_test
+cp ./scripts/deploy-helm.sh ./e2e_test/deploy-helm.sh
+cp -r ./k8s/helm-chart/dynatrace-gcp-function/ ./e2e_test/dynatrace-gcp-function/
+
+VALUES_FILE="./e2e_test/dynatrace-gcp-function/values.yaml"
+
+sed -i '/^gcpProjectId:/c\gcpProjectId: "'${GCP_PROJECT_ID}'"' ${VALUES_FILE}
+sed -i '/^deploymentType:/c\deploymentType: "'${DEPLOYMENT_TYPE}'"' ${VALUES_FILE}
+sed -i '/^dynatraceAccessKey:/c\dynatraceAccessKey: "'${DYNATRACE_ACCESS_KEY}'"' ${VALUES_FILE}
+sed -i '/^dynatraceLogIngestUrl:/c\dynatraceLogIngestUrl: "'${DYNATRACE_LOG_INGEST_URL}'"' ${VALUES_FILE}
+sed -i '/^dynatraceUrl:/c\dynatraceUrl: "'${DYNATRACE_URL}'"' ${VALUES_FILE}
+sed -i '/^logsSubscriptionId:/c\logsSubscriptionId: "'${PUBSUB_SUBSCRIPTION}'"' ${VALUES_FILE}
+sed -i '/^requireValidCertificate:/c\requireValidCertificate: "false"' ${VALUES_FILE}
+
+gcloud container clusters get-credentials "${K8S_CLUSTER}" --region us-central1 --project ${GCP_PROJECT_ID}
+
+cd ./e2e_test || exit 1
+chmod +x ./deploy-helm.sh
+./deploy-helm.sh --service-account e2e-test-gcp-function-sa --role-name e2e_test_gcp_function --quiet || exit 1
```

Comment:
I'm wondering if we shouldn't create unique `service-account` and `role` for each pipeline run (e.g. by adding build id as suffix to names). With fixed names, pipeline will create them only on the first run, and confirm they exist on proceeding runs. This way we are not testing the most crucial path - creating these resources.

Reply:
I agree, but there this need to be carefully checked whether there is a ris on hitting GCP limits.
I'm mostly concerned on custom roles hard limit which is 300. Need to verify if custom roles in deleted state are counted against limit.

Reply:
According to this doc: https://cloud.google.com/iam/docs/creating-custom-roles#deleting-custom-role deleted role is not counted toward the limit

Reply:
after 7 days ;)
"After 7 days, the role is scheduled for permanent deletion. At this point, the role no longer counts towards the limit of 300 custom roles per organization or 300 custom roles per project."
Still we should be safe as build will run only for PRs and in master